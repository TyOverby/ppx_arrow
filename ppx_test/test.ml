let s = [%arrow (fun a -> 
 let x = a + 1 in 
 let y = a + x in  
 let f = [%arrow (fun a -> a)] in
 let z = a + x + y in
 z)] ;; 

