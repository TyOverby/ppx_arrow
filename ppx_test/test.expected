let s =
  compose
    (compose
       (compose
          (compose
             (compose (pure (fun a -> a))
                (pure (fun (a) -> let x = a + 1 in (a, x))))
             (pure (fun (a, x) -> let y = a + x in (a, x, y))))
          (pure
             (fun (a, x, y) ->
                let f = compose (pure (fun a -> a)) (pure (fun (a) -> a)) in
                (a, f, x, y))))
       (pure (fun (a, f, x, y) -> let z = (a + x) + y in (a, f, x, y, z))))
    (pure (fun (a, f, x, y, z) -> z))
